shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/communication/DatabaseCommunicationEngine.java

```diff
@@ -114,24 +113,23 @@ private void prepareCursorStatementContext(final CursorAvailable statementContex
             prepareCursorStatementContext(statementContext, connectionSession, cursorName);
         }
         if (statementContext instanceof CloseStatementContext && ((CloseStatementContext) statementContext).getSqlStatement().isCloseAll()) {
-            FetchOrderByValueGroupsHolder.remove();
-            connectionSession.getCursorDefinitions().clear();
+            connectionSession.getSqlSession().clearCursorSQLSession();
         }
     }
     
     private void prepareCursorStatementContext(final CursorAvailable statementContext, final ConnectionSession connectionSession, final String cursorName) {
         if (statementContext instanceof CursorStatementContext) {
-            connectionSession.getCursorDefinitions().put(cursorName, (CursorStatementContext) statementContext);
+            connectionSession.getSqlSession().getCursorSQLSession().getCursorDefinitions().put(cursorName, (CursorStatementContext) statementContext);
         }
         if (statementContext instanceof CursorDefinitionAware) {
-            CursorStatementContext cursorStatementContext = connectionSession.getCursorDefinitions().get(cursorName);
+            CursorStatementContext cursorStatementContext = (CursorStatementContext) connectionSession.getSqlSession().getCursorSQLSession().getCursorDefinitions().get(cursorName);
             Preconditions.checkArgument(null != cursorStatementContext, "Cursor %s does not exist.", cursorName);
             ((CursorDefinitionAware) statementContext).setUpCursorDefinition(cursorStatementContext);
         }
         if (statementContext instanceof CloseStatementContext) {
-            FetchOrderByValueGroupsHolder.getOrderByValueGroups().remove(cursorName);
-            FetchOrderByValueGroupsHolder.getMinGroupRowCounts().remove(cursorName);
-            connectionSession.getCursorDefinitions().remove(cursorName);
+            connectionSession.getSqlSession().getCursorSQLSession().getOrderByValueGroups().remove(cursorName);
```

Comment:
Can we merge these method into one remove method?

Reply:
Thanks. Done.